### PR TITLE
Refactoring Joiner testcases

### PR DESCRIPTION
**Pull request by CuriousVini**
Issue: https://issues.cask.co/browse/HYDRA-308
Build: http://builds.cask.co/browse/HYP-DBT47-2

### CODE REVIEW

File: core-plugins/src/test/java/co/cask/hydrator/plugin/batch/joiner/JoinerTestRun.java [hunk 1]
```diff
@@ -612,64 +497,81 @@ private void verifyOuterJoinWithoutRequiredInputs(Schema outputSchema, TimeParti
...
+                                                  getBobRecord2(avroOutputSchema),
+                                                  getAlexRecord1(avroOutputSchema),
+                                                  getAlexRecord2(avroOutputSchema),
+                                                  getCathieRecord1(avroOutputSchema),
+                                                  getAvatarRecord1(avroOutputSchema),
+                                                  humtumRecord, sultanRecord);
+    Assert.assertEquals(expected, actual);
+  }
+
+  GenericRecord getBobRecord1(org.apache.avro.Schema avroOutputSchema) {
```
Comment (albertshau) [inline core-plugins/src/test/java/co/cask/hydrator/plugin/batch/joiner/JoinerTestRun.java]: these can be constants right? or do we need new instances of them for some reason?

**Comment by albertshau** [comment]
just one question, otherwise lgtm

File: core-plugins/src/test/java/co/cask/hydrator/plugin/batch/joiner/JoinerTestRun.java (hunk 1 above)
Comment (CuriousVini) [inline core-plugins/src/test/java/co/cask/hydrator/plugin/batch/joiner/JoinerTestRun.java]: As we use different `avroOutputSchema` depending on type of the join (For outer joins, we use nullable fields), we need new instances of records.

**Comment by CuriousVini** [comment]
Build is green. Merging.

**Pull request merged**